dataactbroker/scripts/update_historical_duns.py

```diff
@@ -47,24 +47,19 @@ def clean_duns_csv_data(data):
     return clean_data(data, DUNS, column_mappings, {})
 
 
-def run_duns_batches(file, sess, block_size=10000, batch=0):
-    """Updates DUNS table in batches from csv file"""
+def run_duns_batches(file, sess, block_size=10000):
+    """Updates DUNS table in chunks from csv file"""
     logger.info("Retrieving total rows from duns file")
     start = datetime.now()
     row_count = len(pd.read_csv(file, skipinitialspace=True, header=None, encoding='latin1', quotechar='"',
                                 dtype=str, names=column_headers, skiprows=1))
     logger.info("Retrieved row count of {} in {} s".format(row_count, (datetime.now()-start).total_seconds()))
 
-    batches = row_count // block_size
+    duns_reader_obj = pd.read_csv(file, skipinitialspace=True, header=None,  encoding='latin1', quotechar='"',
+                          dtype=str, names=column_headers, iterator=True, chunksize=block_size, skiprows=1)
 
-    while batch <= batches:
-        logger.info("Begin updating duns batch {} ".format(batch))
+    for duns_df in duns_reader_obj:
```

Comment:
For my understanding, `duns_reader_obj` is an iterator over the CSV file, where each `duns_df` is a chunk of data from the CSV where the chunk size = `block_size`. Is that a correct understanding?

Reply:
Yes that's correct, Pandas returns a `TextFileReader object` to iterate, and each chuck is a dataframe.